### PR TITLE
[Adreno] Adapt reduction schedule for adreno

### PR DESCRIPTION
**Pull request by elvin-n**
Origin cuda schedule uses rfactor that is 10x-50x slower on Adreno than without barries

for example mean on QHD picture on Snapdragon 888 with cuda schedule is executed for 69ms while with new proposed schedule is executed for 6.2
the same for argmin: 183ms -> 3.9ms

### CODE REVIEW

File: python/tvm/relay/op/strategy/adreno.py [hunk 1]
```diff
@@ -208,6 +208,13 @@ def schedule_injective_adreno(attrs, outs, target):
         return topi.adreno.schedule_injective(outs)
 
 
+@schedule_reduce.register(["adreno"])
+def schedule_reduce_adreno(attrs, outs, target):
+    """schedule reduction ops for cuda"""
```
[the remark this answers is not in the document]
Comment (elvin-n) [reply python/tvm/relay/op/strategy/adreno.py]: fixed

File: python/tvm/topi/adreno/reduction.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        return False
+
+    for s in sch.stages:
+        if not s.is_output and isinstance(s.op, tvm.te.ComputeOp):
+            if is_scheduled(s) or len(s.op.reduce_axis) != 0:
+                return False
+    return True
+
+
+def schedule_reduce(outs):
```
[the remark this answers is not in the document]
Comment (elvin-n) [reply python/tvm/topi/adreno/reduction.py:70]: re-used

File: python/tvm/topi/cuda/reduction.py [hunk 3]
```diff
@@ -116,7 +116,7 @@ def is_scheduled(stage):
     return True
 
 
-def schedule_reduce(outs):
+def schedule_reduce_impl(outs, schedule_reduce_stage, schedule_injective_stage):
     """Schedule for inject->reduce->bcast ops.
 
     Parameters
```
Comment (echuraev) [inline python/tvm/topi/cuda/reduction.py:122]: Please add information about new parameters

**Review by echuraev — APPROVED** [review]
LGTM. One minor comment

File: python/tvm/topi/cuda/reduction.py (hunk 3 above)
Comment (elvin-n) [reply python/tvm/topi/cuda/reduction.py:126]: Added

**Review by echuraev — APPROVED** [review]
LGTM. Thanks!

**Comment by elvin-n** [comment]
@masahi could you please review/merge?

**Review by masahi — APPROVED** [review]

**Pull request merged**